### ISSUE TITLE
How to import a function from @/core/content/util.ts in an external program?

### ISSUE THREAD

**Issue by a-github-account** [web-scrobbler/web-scrobbler#3878]
Hello, I'd like to apply the processYtVideoTitle function from @/core/content/util.ts on every element of a specific column of a CSV file. I tried to just clone this repository, move my JS file to the root folder (maybe it's an issue with mixing Typescript with Javascript, but the error I'm getting indicates that this isn't the issue), and use `import { processYtVideoTitle } from "@/core/content/util.ts"`, but this gives me an `error TS2307: Cannot find module '@/core/content/util.ts' or its corresponding type declarations.`

Replacing every @ and # with the actual root directory would be time consuming but possible, but I'm wondering if there's a better way to import the processYtVideoTitle function. I tried running tspath in the root directory of the repository, but it also threw a ton of errors.

I'm not very experienced with node or Typescript, so if I'm missing something obvious here I'd like to be pointed in the right direction.

**Comment by inverse** [comment]
We also provide the most of the filtering in a standalone lib.

See: https://github.com/web-scrobbler/metadata-filter

**Comment by yayuyokitano** [comment]
> We also provide the most of the filtering in a standalone lib.
> 
> See: https://github.com/web-scrobbler/metadata-filter

Yes, though this does not include the title filtering for separating artist/track names within a title, it only includes filtering within specific fields. I think I have talked about wanting to separate this out at some point in the future but rn some of the code is tied exclusively to util.ts.

Your best bet for using this within a javascript project is probably to isolate just the processYtVideoTitle function and the functions it depends on into its own file, and converting it to JS, either manually by removing type annotations or automatically by running it through tsc, and just using it as raw JS.

**Comment by yayuyokitano** [comment]
Though to add, the reason you're getting the error you're getting is that @/core/content/util is referring to a specific filepath relative to an alias.

We have the [alias @ set up in our tsconfig](https://github.com/web-scrobbler/web-scrobbler/blob/bdeeaa8e02a93a92b63328bbf9a38a430b06e48a/tsconfig.json#LL11C7-L11C7) to be the src directory of the project, and it then goes from that to core to content to util.ts

**Comment by a-github-account** [comment]
Thanks for the responses! I'll try just using tsc and copy-pasting all the functions.
